custom_components/homewhiz/homewhiz.py

```diff
@@ -89,6 +89,15 @@ def clamp(value: int):
     return value if value < 128 else value - 128
 
 
+def runningDelay( first_byte, second_byte ):
+    if first_byte == 128:
+        return None
+    elif (first_byte * 60 + second_byte) < 7808:
+        return first_byte * 60 + second_byte
+    else:
+        return first_byte * 60 + second_byte - 7808
```

Comment:
I think it boils down to `clamp(first_byte) + second_byte`

Can you write a test for it?
You can pretty easily get your actual message payload from the logs and write a simple assertion like here: https://github.com/rowysock/home-assistant-HomeWhiz/blob/e8828c86d2563d7d4fc2061144ed9c5f48d3333a/custom_components/tests/parser.py#L40

Reply:
And when you are at it - we are using black for code formatting, so if you could format your code that would be great 😄

Reply:
@rowysock I think you are right, I will try to do it this weekend. 

@TechHummel I will try to setup some sort of IDE, must be better than editing the code straight off the running HA machine over ssh using nano* and copy pasting the result on github... ;) (*Believe me it's even worse)

Reply:
If you want to modify the code on your running machine, there is a ssh pluign avaibale for VSCode.

Reply:
@maxx4096 I recommend setting up Home assistant on your PC. It is as simple as setting up python and virtual env, `pip install homeassistant`, and running ./bin/hass. The local instance is empty so it starts instantaneously. it saves me a lot of time

Reply:
I spun a new VM with VSCode. Testing locally is not really possible as my PC is too far away to get a Bluetooth connection (unless I'm missing something).

After messing around with VSCode and Git, I ended up creating a new branch, discarding the new function and rewriting what was already there using the clamp() function as you suggested, added tests for delay selected but not yet started and delay started. 
I also ran black and I'm not liking the result as it broke down an inline if into multiple lines... Any option I have to set?

While I wrote the tests I feel stupid not knowing how to run them and verify they actually pass...
Thanks

Reply:
to run the tests locally first install the test dependencies: `pip install -r requirements_test.txt` and then just run `pytest`

Reply:
If that works for you I will add a readme section on how to run tests

Reply:
I did not manage to run pytest without renaming parser.py to test_parser.py and reverting afterwards...

As for the readme I wrote this https://github.com/maxx4096/home-assistant-HomeWhiz/blob/Documentation/linux_dev.md .

Reply:
Are you sure you have the latest version checked out? 
I've fixed the tests and renamed the `parser.py` to `test_parser.py` a few days ago

Reply:
@maxx4096 This is one of the best guides I've ever seen!

Reply:
I'm new to git, and I did not refresh the thing while I was working on it. Seeing how many PR got merged in the interval, that must be why I couldn't run the tests. I really need to learn how to manage my workflow in an active project...

As for the doc, this is my attempt to save someone else's sanity. I went too fast, ended up making a lot of mistakes and loosing hours of my time because my env wasn't set-up properly and had to redo everything. Right this time (I hope)